cmd/apps/nfs_app.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallNfsProvisioner() *cobra.Command {
+	var nfsProvisionerApp = &cobra.Command{
+		Use:          "nfs-client-provisioner",
+		Short:        "Install nfs client provisioner",
+		Long:         "Install nfs client provisioner to create dynamic persistent volumes",
+		Example:      "arkade install nfs-client-provisioner --nfs-server=x.x.x.x --nfs-path=/exported/path",
+		SilenceUsage: true,
+	}
+
+	nfsProvisionerApp.Flags().StringP("namespace", "n", "default", "The namespace to install nfs-client (default: default")
+	nfsProvisionerApp.Flags().String("nfs-server", "", "")
+	nfsProvisionerApp.Flags().String("nfs-path", "", "")
+	nfsProvisionerApp.Flags().Bool("update-repo", true, "Update the helm repo")
+	nfsProvisionerApp.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set =true)")
+
+	nfsProvisionerApp.RunE = func(command *cobra.Command, args []string) error {
+		helm3 := true
+
+		namespace, _ := nfsProvisionerApp.Flags().GetString("namespace")
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		clientArch, clientOS := env.GetClientArch()
+
+		log.Printf("Client: %s, %s\n", clientArch, clientOS)
+
+		log.Printf("User dir established as: %s\n", userPath)
+
+		if err := os.Setenv("HELM_HOME", path.Join(userPath, ".helm")); err != nil {
+			return err
+		}
+
+		nfsServer, _ := command.Flags().GetString("nfs-server")
+		nfsPath, _ := command.Flags().GetString("nfs-path")
+
+		if len(nfsServer) == 0 {
+			return fmt.Errorf("--nfs-server required")
+		}
+
+		if len(nfsPath) == 0 {
+			return fmt.Errorf("--nfs-path required")
+		}
+
+		overrides := map[string]string{}
+
+		overrides["nfs.server"] = nfsServer
+		overrides["nfs.path"] = nfsPath
+
+		switch clientArch {
+		case "arm", "armhf", "arm64", "aarch64":
+			overrides["image.repository"] = "quay.io/external_storage/nfs-client-provisioner-arm"
+		}
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		if err := config.MergeFlags(overrides, customFlags); err != nil {
+			return err
+		}
+
+		nfsProvisionerOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmPath(path.Join(userPath, ".helm")).
+			WithHelmRepo("stable/nfs-client-provisioner").
+			WithHelmURL("https://kubernetes-charts.storage.googleapis.com").
+			WithOverrides(overrides)
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeconfigPath, _ := command.Flags().GetString("kubeconfig")
+			nfsProvisionerOptions.WithKubeconfigPath(kubeconfigPath)
+		}
+
+		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS, helm3)
+		if err != nil {
+			return err
+		}
+
+		_, err = apps.MakeInstallLoki(nfsProvisionerOptions)
```

Comment:
Can we rename the `MakeInstallLoki` function (and rename the file) to `MakeInstallChart` and `chart_app.go`

It should have been generic when I made it but I didn't realize until I was working on something else

Reply:
I'm not a fan of changing different things on the same PR but as it seems we all agree on this one, let's make an exception :)